owncloudData/src/test/java/com/owncloud/android/data/transfers/implementation/OCLocalTransferDataSourceTest.kt

```diff
@@ -0,0 +1,326 @@
+
+        val transferSucceeded = ocTransfer.copy()
+        transferSucceeded.status =  TransferStatus.TRANSFER_SUCCEEDED
+
+        val transferListOrdered = listOf(ocTransfer, transferQueue, transferFailed, transferSucceeded)
+
+        every {
+            transferDao.getAllTransfersAsStream()
+        } returns flowOf(transferListRandom)
+
+        val actualResult = ocLocalTransferDataSource.getAllTransfersAsStream().first().map { it }
+
+        assertEquals(transferListOrdered, actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getAllTransfersAsStream()
+        }
+    }
+
+
+    @Test
+    fun `getLastTransferFor returns a OCTransfer`() {
+
+        every {
+            transferDao.getLastTransferWithRemotePathAndAccountName(ocTransfer.remotePath, OC_ACCOUNT_NAME)
+        } returns transferEntity
+
+        val actualResult = ocLocalTransferDataSource.getLastTransferFor(ocTransfer.remotePath, OC_ACCOUNT_NAME)
+
+        assertEquals(transferEntity.toModel(), actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getLastTransferWithRemotePathAndAccountName(ocTransfer.remotePath, OC_ACCOUNT_NAME)
+        }
+    }
+
+    @Test
+    fun `getCurrentAndPendingTransfers returns a list of OCTransfer`() {
+
+        every {
+            transferDao.getTransfersWithStatus(listOf(TransferStatus.TRANSFER_IN_PROGRESS.value, TransferStatus.TRANSFER_QUEUED.value))
+        } returns listOf(transferEntity)
+
+        val actualResult = ocLocalTransferDataSource.getCurrentAndPendingTransfers()
+
+        assertEquals(listOf(transferEntity.toModel()), actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getTransfersWithStatus(listOf(TransferStatus.TRANSFER_IN_PROGRESS.value, TransferStatus.TRANSFER_QUEUED.value))
+        }
+    }
+
+    @Test
+    fun `getFailedTransfers returns a list of OCTransfer`() {
+
+        every {
+            transferDao.getTransfersWithStatus(listOf(TransferStatus.TRANSFER_FAILED.value))
+        } returns listOf(transferEntity)
+
+        val actualResult = ocLocalTransferDataSource.getFailedTransfers()
+
+        assertEquals(listOf(transferEntity.toModel()), actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getTransfersWithStatus(listOf(TransferStatus.TRANSFER_FAILED.value))
+        }
+    }
+
+    @Test
+    fun `getFinishedTransfers returns a list of OCTransfer`() {
+
+        every {
+            transferDao.getTransfersWithStatus(listOf(TransferStatus.TRANSFER_SUCCEEDED.value))
+        } returns listOf(transferEntity)
+
+        val actualResult = ocLocalTransferDataSource.getFinishedTransfers()
+
+        assertEquals(listOf(transferEntity.toModel()), actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getTransfersWithStatus(listOf(TransferStatus.TRANSFER_SUCCEEDED.value))
+        }
+    }
+
+    @Test
+    fun `clearFailedTransfers clears it correctly`() {
```

Comment:
```suggestion
    fun `clearFailedTransfers removes transfers correctly`() {
```
Try not to use the same words as in the method name

Reply:
Done